app/components/message_attachments/action_menu/action_menu.js

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import {Text, View} from 'react-native';
+import PropTypes from 'prop-types';
+import {intlShape} from 'react-intl';
+import Icon from 'react-native-vector-icons/FontAwesome';
+
+import FormattedText from 'app/components/formatted_text';
+import {preventDoubleTap} from 'app/utils/tap';
+import {makeStyleSheetFromTheme, changeOpacity} from 'app/utils/theme';
+
+export default class ActionMenu extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            doPostAction: PropTypes.func.isRequired,
+            setMenuActionSelector: PropTypes.func.isRequired,
+        }).isRequired,
+        id: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired,
+        dataSource: PropTypes.string,
+        options: PropTypes.arrayOf(PropTypes.object),
+        postId: PropTypes.string.isRequired,
+        theme: PropTypes.object.isRequired,
+        navigator: PropTypes.object,
+    };
+
+    static contextTypes = {
+        intl: intlShape,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedText: null,
+        };
+    }
+
+    handleSelect = (selected) => {
+        if (!selected) {
+            return;
+        }
+
+        const {dataSource, actions, postId, id} = this.props;
+
+        let selectedText;
+        let selectedValue;
+        if (dataSource === 'users') {
+            selectedText = selected.username;
+            selectedValue = selected.id;
+        } else if (dataSource === 'channels') {
+            selectedText = selected.display_name;
+            selectedValue = selected.id;
+        } else {
+            selectedText = selected.text;
+            selectedValue = selected.value;
+        }
+
+        this.setState({selectedText});
+
+        actions.doPostAction(postId, id, selectedValue);
+    }
+
+    goToMenuActionSelector = preventDoubleTap(() => {
+        const {intl} = this.context;
+        const {navigator, theme, actions, dataSource, options} = this.props;
+
+        actions.setMenuActionSelector(dataSource, this.handleSelect, options);
+
+        navigator.push({
```

Comment:
why a push and not a modal?

Reply:
I copied how the add channel member screen worked. If there's a better way to do this let me know and I can update it, I'm not familiar with the patterns used on mobile

Reply:
This is the way Slack does it and I was told to copy that.